src/EditorFeatures/Core/StackTraceExplorer/AbstractStackTraceSymbolResolver.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Immutable;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.StackTraceExplorer;
+using Microsoft.CodeAnalysis.EmbeddedLanguages.StackFrame;
+using Microsoft.CodeAnalysis.LanguageServices;
+using Microsoft.CodeAnalysis.Shared.Extensions;
+
+namespace Microsoft.CodeAnalysis.Editor.StackTraceExplorer
+{
+    internal abstract class AbstractStackTraceSymbolResolver
+    {
+        public abstract Task<IMethodSymbol?> TryGetBestMatchAsync(Project project, INamedTypeSymbol type, StackFrameSimpleNameNode methodNode, StackFrameParameterList methodArguments, StackFrameTypeArgumentList? methodTypeArguments, CancellationToken cancellationToken);
```

Comment:
consider wrapping. i like the general rule of "should not have to horizontal scroll on github" :)

Reply:
> "should not have to horizontal scroll on github"

That's very dependent on monitor size. I have a very wide monitor :)